docker/index.js

```diff
@@ -2,100 +2,241 @@ const express = require('express');
+	const inputPath = transaction.tx_body.operation.ref;
+	const parsedInputPath = parsePath(inputPath);
+
+	// pre-check the output path
+	const outputPath = await formatPath([...parsedInputPath.slice(0, parsedInputPath.length - 1), "signed_data"]);
+	const errorPath = formatPath([...parsedInputPath.slice(0, parsedInputPath.length - 1), "error"]);
+
+	console.log(outputPath);
+	// init pinata sdk
+	const pinata = new pinataSDK({ pinataApiKey, pinataSecretApiKey });
+
+
+	// get generated ainft image url with with task id
+	const result = await axios.get(`${process.env.SD_INPAINTING_ENDPOINT}/tasks/${taskId}`);
+
+	const imageUrls = result.data.result;
+
+	// get image file from url
+	const imageDataResponse = await axios.get(imageUrls[value.params.index || 1], {
+		responseType: "arraybuffer",
+	});
+
+	// for test
+	const imageBuffer = fs.readFileSync(__dirname + "/test.png");
+
+	// image file to readable stream
+	const imageDataStream = new Readable();
+	imageDataStream.push(imageBuffer);
+	imageDataStream.push(null);
+
+	// upload image to pinata
+	const options = {
+		pinataMetadata: {
+			name: `${taskId}_image`,
+		},
+		pinataOptions: {
+			cidVersion: 0,
+		},
+	};
+
+	// upload image to ipfs
+	const uploadImgRes = await pinata.pinFileToIPFS(imageDataStream, options)
+		.catch(err => {
+			ain.db.ref(errorPath).setValue({
+				value: {
+					state:"Error",
+					msg:"Image upload fail. check your inforamtion of Image"
+				},
+			})
+			.then(res => console.log(res))
+			.catch((e) => {
+				console.error(`setValue failure:`, e);
+			});
+		});
+
+	// get origin metadata
+	const originMetadata = await axios.get(`https://gateway.pinata.cloud/ipfs/${value.contract_info.metadata_cid}/${value.contract_info.token_id}`, {
+		headers: {
+			'Accept': '*/*'
+		}
+	});
+
+	// metadata will be writed in ipfs
+	const metadata = {
+		attributes: originMetadata.attributes,
+		description: originMetadata.description,
+		image: originMetadata.image,
+		name: originMetadata.name,
+		namespaces: {
+			ainetwork: {
+				ain_tx: transaction.hash, // need 
+				old_metadata: value.contract_info.metadata_cid, // do not apply yet
+				updated_at: Date.now()
+			},
+		}
+	}
+	const metadataOption = {
+		pinataMetadata: {
+			name: `${taskId}_metadata`,
+		},
+		pinataOptions: {
+			cidVersion: 0,
+		},
+	}
+
+	try{
+		// upload metadata to ipfs
+		uploadMetadataRes = await pinata.pinJSONToIPFS(metadata, metadataOption);
+	}
+	catch (e){
+		// if fail upload metadata, uploaded image is unpined in pinata.
+		console.error(e);
+		await pinata.unpin(uploadImgRes.IpfsHash);
+		await ain.db.ref(errorPath).setValue({
+			value: {
+				name:"ainftize",
+				state:"Error",
+				msg:"Metadata upload fail. check your inforamtion of metadata"
+			},
+		})
+		.then(res => console.log(res))
+		.catch((e) => {
+			console.error(`setValue failure:`, e);
+		});
+
+		return;
+	}
+
+	await ain.db.ref(outputPath).setValue({
+		value: {
+			metadata_cid: uploadMetadataRes.IpfsHash,
```

Comment:
should follow "AIN_MetadataV2" described in following documents: https://www.notion.so/comcom/About-Metadata-cd48a6393ef948f097388ef5895296e7